backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ir/interop/IrProviderForCEnumAndCStructStubs.kt

```diff
@@ -67,12 +72,13 @@ internal class IrProviderForCEnumAndCStructStubs(
             if (field != null)
                 error("Module has already been set")
             field = value
-            value.files += filesMap.values
+            value.files += filesMap.filterKeys { moduleFilter(it.module) }.values
         }
 
     fun canHandleSymbol(symbol: IrSymbol): Boolean {
         if (!symbol.isPublicApi) return false
         if (symbol.signature.run { !IdSignature.Flags.IS_NATIVE_INTEROP_LIBRARY.test() }) return false
+        if (!moduleFilter(symbol.descriptor.module)) return false
```

Comment:
Do I understand correctly that this is an optimization to use lazy IR instead of generated IR for declarations from cached interop libraries?

Reply:
Yes

Reply:
AFAIK, lazy IR is harmful, because it potentially generates unbound symbols after psi2ir.
Is this consideration applicable here?

Reply:
Discussed offline. Should be ok.

Reply:
However it makes sense to add a brief comment mentioning that this line makes IR machinery fall back to lazy IR for this declaration.